src/reactor/mod.rs

```diff
@@ -73,7 +74,13 @@ struct Inner {
 /// and will instead use and implicitly configured handle for your thread.
 #[derive(Clone)]
 pub struct Handle {
-    inner: Weak<Inner>,
+    repr: HandleRepr,
+}
+
+#[derive(Clone)]
+enum HandleRepr {
```

Comment:
I think this enum should be avoidable if you can create a `HelperThread` instance without actually starting the thread. You would get the `Handle` of the reactor that hasn't actually started running. If creating a reactor fails, you could set `inner` to `Weak::new()`.

Either way, I think being able to get the handle to the global reactor when the thread isn't running is useful in its own right. This goes back to the questions I posed in the above comment.